src/lib.rs

```diff
@@ -100,6 +164,7 @@ impl<'de> Deserialize<'de> for UrlWrapper {
 
 #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Debug, Serialize, Deserialize)]
 pub enum Source {
+    Registry(Registry, String),
     CratesIo(String),
```

Comment:
I think I'd prefer if these were collapsed into `Registry`, `CratesIo` is just a specific `Registry`.